src/pages/tabs.astro

```diff
@@ -0,0 +1,296 @@
+
+      Bound to the tablist, this handles keyboard accessibility for selecting tabs.
+
+      ## Dependencies
+
+      ### `aria-selected:` variant
+
+      Styling the selected tab is easiest using the `aria-selected:` variant:
+
+      ```js
+      const plugin = require('tailwindcss/plugin')
+
+      module.exports = plugin(({ addVariant, e }) => {
+        addVariant('aria-selected', ({ modifySelectors, separator }) => {
+          modifySelectors(({ className }) => {
+            return `[aria-selected='true'] .${e(
+              `aria-selected${separator}${className}`
+            )}, [aria-selected='true'].${e(`aria-selected${separator}${className}`)}`
+          })
+        })
+      })
+      ```
+    </Markdown>
+  </section>
+
+  <section class="divide-y divide-slate-700" slot="demos">
+    <CodeDemo heading="Basic" className="py-8">
+      <div slot="preview">
+        <div
+          class="max-w-md mx-auto"
+          data-controller="tabs"
+        >
+          <ul
+            class="flex justify-center mb-4"
+            data-action="keyup->tabs#cycleTabs"
+            role="tablist"
+          >
+            {['Tab 1', 'Tab 2', 'Tab 3'].map((text, i) => (
+              <li>
+                <button
+                  data-tabs-target="tab"
+                  data-action="tabs#selectTab"
+                  class={cx(
+                    'py-2 px-5 tracking-wide font-semibold border-b-2 border-white border-opacity-20',
+                    'hover:bg-white hover:bg-opacity-10',
+                    'aria-selected:border-opacity-100'
+                  )}
+                  aria-selected={i === 0 ? 'true' : null}
+                  tabindex={i !== 0 ? '-1' : null}
+                >
+                  {text}
+                </button>
+              </li>
+            ))}
+          </ul>
+
+          <div
+            class="p-12 border-2 border-white border-opacity-20 rounded-md shadow-lg text-6xl text-center font-extrabold"
+            data-tabs-target="panel"
+            role="tabpanel"
+          >
+            Panel 1
+          </div>
+          <div
+            class="p-12 border-2 border-white border-opacity-20 rounded-md shadow-lg text-6xl text-center font-extrabold"
+            data-tabs-target="panel"
+            role="tabpanel"
+            hidden
+          >
+            Panel 2
+          </div>
+          <div
+            class="p-12 border-2 border-white border-opacity-20 rounded-md shadow-lg text-6xl text-center font-extrabold"
+            data-tabs-target="panel"
+            role="tabpanel"
+            hidden
+          >
+            Panel 3
+          </div>
+        </div>
+      </div>
+
+      <div slot="code">
+        <Markdown>
+          ```jsx
+          <div data-controller="tabs">
+            <ul
+              class="flex"
+              data-action="keyup->tabs#cycleTabs"
+              role="tablist"
+            >
+              {['Tab 1', 'Tab 2', 'Tab 3'].map((text, i) => (
```

Comment:
Super minor, but the JSX here might potentially trip up newer folks who aren't familiar what parts are Stimulus syntax and what's JSX.

Reply:
fair, i struggled with that. should we commit to twig demo code? the hold up for me was i couldnt actually (or easily) verify the twig code the way i could the jsx code

Reply:
oh and then i thought i could just manually do multiple elements but i figured the loop logic was helpful for the attributes even if it was in jsx 🤷

Reply:
somewhat related, entirely nit/subjective: did you consider having everything in a single array? like

```
tabsAndTabpanels = [
  {
    tabContent: …,
    tabpanelContent: …,
  },
  …
]
```